### PR TITLE
add configFile reference to alertmanager

### PR DESCRIPTION
**Pull request by lswith**
This addresses https://github.com/coreos/prometheus-operator/issues/2003

### CODE REVIEW

**Comment by brancz** [comment]
I think I would prefer just `config` or `configSecret`, otherwise this just needs a `make generate-in-docker` :slightly_smiling_face: .

**Comment by lswith** [comment]
Updated to use `configSecret`

File: pkg/client/monitoring/v1/types.go [hunk 1]
```diff
@@ -629,6 +629,8 @@ type AlertmanagerSpec struct {
 	// object, which shall be mounted into the Alertmanager Pods.
 	// The Secrets are mounted into /etc/alertmanager/secrets/<secret-name>.
 	Secrets []string `json:"secrets,omitempty"`
+	// ConfigSecret is the secret file to use for the alertmanager configuration
```
Comment (mxinden) [inline pkg/client/monitoring/v1/types.go]: `s/secret file/Kubernetes secret/` would read better to me, what do you think?

**Comment by lnovara** [comment]
@lswith do you have any updates about this PR? 

We are Kustomize heavy users and it would be really nice to have Alertmanager get secrets in a more transparent way.

Let me know if I can be of any help.

Thanks.

**Comment by brancz** [comment]
For what its worth we're happy to review a PR if anyone else wants to pick this up! :slightly_smiling_face:

**Pull request closed without merging**